aws/cloudformation/bootstrap_old.sh.erb

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash -x
+# UserData bootstrap script for CloudFormation stack instances.
+# Note: Minimize changes to this script! Every time its contents change,
+# the next CloudFormation update will reboot all servers using this script.
+
+# Bootstrap prerequisites
+apt-get update
+apt-get -y install python3-pip curl
+test "$(pip3 show awscli)" || pip3 install awscli
+
+STACK=<%=stack_name%>
+REGION=<%=region%>
+
+function ref() {
+  aws cloudformation describe-stack-resources \
+    --stack-name=${STACK} \
+    --logical-resource-id=$1 \
+    --output=text \
+    --region ${REGION} \
+    | cut -f3
+}
+
+S3_BUCKET=<%=s3_bucket%>
+ENVIRONMENT=<%=environment%>
+RUN_LIST='<%=run_list.to_json%>'
+INSTANCE_ID=$(curl -s http://169.254.169.254/latest/meta-data/instance-id)
+NODE_NAME=<%=node_name%>
+RESOURCE_ID=<%=resource_id%>
+
+FIRST_BOOT=/etc/chef/first-boot.json
+mkdir -p $(dirname ${FIRST_BOOT})
+cat <<JSON > ${FIRST_BOOT}
+{
+  "cdo-repository": {
+    "revision": "<%=commit%>",
+    "sync": true
+  },
+  "run_list": ${RUN_LIST}
+}
+JSON
+
+# Bootstrap the first cdo-apps Rakefile build on a new instance.
+# TODO move this run-once notification somewhere more appropriate
+rm -f /etc/init.d/{dashboard,pegasus}
+
+OPTIONS="<%=local_mode ? '-z' : ''%> -n ${NODE_NAME} -r '${RUN_LIST}' -e ${ENVIRONMENT}"
+sudo -u ubuntu bash -c "aws s3 cp s3://${S3_BUCKET}/chef/bootstrap-${STACK}.sh - | sudo bash -s -- ${OPTIONS}"
+
+[ $? -eq 0 ] && STATUS=SUCCESS || STATUS=FAILURE
+[ ${STATUS} = 'SUCCESS' ] && LIFECYCLE_RESULT=CONTINUE || LIFECYCLE_RESULT=ABANDON
+
+# Signal CompleteLifecycleAction, in case this instance was launched from an Auto Scaling process.
+LIFECYCLE_HOOK=$(ref WebServerHook)
+AUTO_SCALING_GROUP=$(ref WebServer)
+if [ -n "${LIFECYCLE_HOOK}" ] && [ -n "${AUTO_SCALING_GROUP}" ]; then
+  aws autoscaling complete-lifecycle-action \
+    --lifecycle-action-result ${LIFECYCLE_RESULT} \
+    --instance-id ${INSTANCE_ID} \
+    --lifecycle-hook-name ${LIFECYCLE_HOOK} \
+    --auto-scaling-group-name ${AUTO_SCALING_GROUP} \
+    --region ${REGION} \
+    || true
+fi
+
+# Signal CloudFormation, in case this instance was launched from a CloudFormation stack update.
+aws cloudformation signal-resource \
+  --unique-id ${INSTANCE_ID} \
+  --stack-name ${STACK} \
+  --logical-resource-id ${RESOURCE_ID} \
+  --status ${STATUS} \
+  --region ${REGION} \
+  || true
```

Comment:
This is a copy of the old `bootstrap_chef_stack.sh.erb`, used to preserve the existing auto scaling group unchanged through the migration. To be removed afterwards.